include/RAJA/util/View.hpp

```diff
@@ -181,11 +186,20 @@ struct MultiView
   template<bool IsConstView = std::is_const<value_type>::value>
   RAJA_INLINE constexpr MultiView(
       typename std::enable_if<IsConstView, NonConstView>::type const& rhs)
-      : layout(rhs.layout),
-        data(rhs.data)
+      : layout(rhs.get_layout()),
+        data(rhs.get_data())
   {}
 
-  RAJA_INLINE void set_data(pointer_type data_ptr) { data = data_ptr; }
+  template<typename... Args>
+  RAJA_HOST_DEVICE RAJA_INLINE void set_layout(Args... dim_sizes) { layout(dim_sizes...); }
+
+  RAJA_HOST_DEVICE RAJA_INLINE void set_layout(layout_type&& ly) { layout(ly); }
```

Comment:
These two will conflict. Are we trying to make a new view here or index into the layout?

Reply:
Are we trying to make a new layout here or index into the layout?

Reply:
Removed the variadic sizes setter because that will require more support in MultiView. We're just trying to set the layout of an emptily-constructed MultiView, so I think it makes more sense to  set the layout with a layout object.